OpenSim/Common/Component.cpp

```diff
@@ -163,12 +163,15 @@ void Component::addComponent(Component* subcomponent)
 void Component::prependComponentPathToConnecteePath(
         Component& subcomponent) {
     const std::string compPath = subcomponent.getAbsolutePathString();
+    const Component& root = subcomponent.getRoot();
     for (auto& comp : subcomponent.updComponentList()) {
         for (auto& it : comp._socketsTable) {
-            it.second->prependComponentPathToConnecteePath(compPath);
+            if (!root.hasComponent(it.second->getConnecteePath()))
+                it.second->prependComponentPathToConnecteePath(compPath);
         }
         for (auto& it : comp._inputsTable) {
-            it.second->prependComponentPathToConnecteePath(compPath);
+            if (!root.hasComponent(it.second->getConnecteePath()))
+                it.second->prependComponentPathToConnecteePath(compPath);
```

Comment:
The connectees of inputs are not components.

Reply:
Thanks @chrisdembia I undid the change for inputs, now all test cases pass.  Thank you :+1